test/cpp/naming/resolver_component_tests_runner_invoker.cc

```diff
@@ -55,6 +55,16 @@ ABSL_FLAG(
     "False indicates that this test is running under run_tests.py. "
     "Child process test binaries are located differently based on this flag. ");
 
+#ifdef GPR_WINDOWS
+ABSL_FLAG(bool, running_locally, false,
+          "This flag only exists on Windows and is only effective when running "
+          "under Bazel. "
+          "Set this to true (e.g. --test_arg=--running_locally=true) when "
+          "running the test locally. "
+          "This is because the RUNFILES dir behavior is different between "
```

Comment:
What exactly is the difference? Is that documented somewhere? Have you investigated whether it's possible to make bazel runfiles behave the same when running locally vs remotely?

Reply:
good question. I found there is a `--enable_runfiles` option and is default off on Windows so Bazel will just create a MANIFEST file under the RUNFILES dir instead of creating the symlinks tree. When enabled, Bazel errored with:
```
C:\users\yijiem\_bazel_yijiem\install\1a52a8611778e900b15612375871f5a1\build-runfiles.exe --allow_relative bazel-out/x64_windows-fastbuild/bin/test/cpp/naming/utils/tcp_connect.exe.runfiles_manifest bazel-out/x64_windows-fastbuild/bin/test/cpp/naming/utils/tcp_connect.exe.runfiles: Process exited with status 1: Process exited with status 1
build-runfiles error: CreateSymbolicLinkW failed:
Bazel needs to create symlinks to build the runfiles tree.
Creating symlinks on Windows requires one of the following:
    1. Bazel is run with administrator privileges.
    2. The system version is Windows 10 Creators Update (1703) or later and developer mode is enabled.
```

After qualifying for these 2 conditions, Bazel runs the test just fine. I think most of the change in this file can be reverted and I have added logging to hint about this flag if Bazel is invoked locally (using the `GRPC_PORT_ISOLATED_RUNTIME` preprocessor symbol check).